tst/com/amazon/corretto/crypto/provider/test/RsaGenTest.java

```diff
@@ -189,19 +189,34 @@ private static void assertConsistency(final RSAPublicKey pub, final RSAPrivateCr
         assertNotNull(priv);
         assertEquals(pub.getPublicExponent(), priv.getPublicExponent());
         assertNotNull(pub.getModulus());
-        assertEquals(pub.getModulus(), priv.getModulus());
+        BigInteger modulus = priv.getModulus();
+        assertEquals(pub.getModulus(), modulus);
         assertNotNull(priv.getPrivateExponent());
-        assertNotNull(priv.getPrimeExponentP());
-        assertNotNull(priv.getPrimeExponentQ());
+        assertNotNull(priv.getPrimeP());
+        assertNotNull(priv.getPrimeQ());
         assertNotNull(priv.getPrimeExponentP());
         assertNotNull(priv.getPrimeExponentQ());
         assertNotNull(priv.getCrtCoefficient());
 
         // Do the underlying math
-        assertEquals(priv.getModulus(), priv.getPrimeP().multiply(priv.getPrimeQ()));
-        assertEquals(priv.getPrivateExponent().mod(priv.getPrimeP().subtract(BigInteger.ONE)), priv.getPrimeExponentP());
-        assertEquals(priv.getPrivateExponent().mod(priv.getPrimeQ().subtract(BigInteger.ONE)), priv.getPrimeExponentQ());
-        assertEquals(priv.getPrimeQ().modInverse(priv.getPrimeP()), priv.getCrtCoefficient());
+        final BigInteger p = priv.getPrimeP();
+        final BigInteger q = priv.getPrimeQ();
```

Comment:
Would it be worth it to also check if `p` and `q` are primes? The following code would check that both are prime, up to 1/(2^128) certainty.
```
assertTrue(p.isProbablePrime(128));
assertTrue(q.isProbablePrime(128));
```
Source: https://docs.oracle.com/javase/7/docs/api/java/math/BigInteger.html#isProbablePrime(int)

Reply:
I like this. and will add it.